### PR TITLE
Adds Github webhook endpoint

### PR DESCRIPTION
**Pull request by tallysmartins**

### CODE REVIEW

File: lib/elixir_bench_web/controllers/webhooks_controller.ex [hunk 1]
```diff
@@ -0,0 +1,62 @@
+defmodule ElixirBenchWeb.WebHooks do
```
Comment (PragTob) [inline lib/elixir_bench_web/controllers/webhooks_controller.ex]: this isn't general webhooks is it, but specifically github right? I think I'd even put that in a github namespace.

File: lib/elixir_bench_web/controllers/webhooks_controller.ex [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  defp process_event("ping", conn, _payload) do
+    conn
+    |> json(%{message: "pong"})
+  end
+
+  defp process_event("pull_request", conn, payload) do
+    # data is fetched from sender's repository
+    slug = payload["pull_request"]["head"]["repo"]["full_name"]
+    branch_name = payload["pull_request"]["head"]["ref"]
+    commit_sha = payload["pull_request"]["head"]["sha"]
```
Comment (PragTob) [inline lib/elixir_bench_web/controllers/webhooks_controller.ex]: you might be able to use helper functions such as `get_in` to make the access nicer here

File: lib/elixir_bench_web/controllers/webhooks_controller.ex [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+         {:ok, job} <-
+           Benchmarks.create_job(repo, %{branch_name: branch_name, commit_sha: commit_sha}) do
+      conn
+      |> render(ElixirBenchWeb.JobView, "show.json", job: job, repo: repo)
+    end
+  end
+
+  defp process_event("push", conn, payload) do
+    conn
+    |> json(%{event: "push", content: payload})
```
Comment (PragTob) [inline lib/elixir_bench_web/controllers/webhooks_controller.ex]: why are we just replying here?

Comment (PragTob) [reply lib/elixir_bench_web/controllers/webhooks_controller.ex]: Shouldn't we be doing _roughly_ the same thing as above aka create a new job to measure the performance impact?

File: test/elixir_bench_web/controllers/webhooks_controller_test.exs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      end
+
+      assert %{"branch_name" => "changes", "repo_slug" => "baxterthehacker/public-repo"} = data
+    end
+
+    defp set_headers(conn, event) do
+      conn
+      |> put_req_header("x-github-event", event)
+    end
+  end
```
Comment (PragTob) [inline test/elixir_bench_web/controllers/webhooks_controller_test.exs]: some mall tests about non happy path behaviour here would be nice

File: test/support/factory.ex [hunk 5]
```diff
@@ -143,4 +143,422 @@ defmodule ElixirBench.Factory do
       }
     }
   end
+
+  def pull_request_payload() do
```
Comment (PragTob) [inline test/support/factory.ex]: a link to where you got this from would be nice imo - also not sure we should/need to keep all of it. I'd also move it out of this file to not clutter the other factory stuff (it's quite huge) and rather move it into its own little github factory file

**Comment by tallysmartins** [comment]
Yes, still some small details to go, see #26. Just finishing the "push" event and I also want to make a manual test. I think the auth verification with ```secret key``` can be added later.

**Comment by PragTob** [comment]
@tallysmartins what do you mean the verification with secret key? If we want to add it later we should definitely create an issue for it lest we forget

**Comment by tallysmartins** [comment]
I mean that GitHub let us set a secret password to Hash the payload and verify its authenticity. I think we will use it when we setup the market place app, but not sure. Anyway we can add it later. Its just a check sum of the payload that comes in the header.

**Comment by PragTob** [comment]
I think it's for us so that not just about anyone can trigger a build and thereby DoS us easily but it should be fine-ish

**Pull request merged**